app/pages/learn/01_tutorial/07_rich_client_apps/02_introduction-javafx-animation.md

```diff
@@ -0,0 +1,265 @@
+---
+id: javafx.animation
+title: Introduction to JavaFX animations
+slug: learn/javafx-animations
+type: tutorial
+group: rich-client-apps
+category: javafx
+category_order: 2
+layout: learn/tutorial.html
+subheader_select: tutorials
+main_css_id: learn
+description: "Learn to create advanced JavaFX animations"
+last_update: 2024-05-17
+---
+
+The [javafx.animation](javafxdoc:AnimationPackageSummary) package offers a simple framework for creating animations and transitions in a JavaFX application.
+It operates on the principle of [WritableValue\<T\>](javafxdoc:WritableValue) which are used across JavaFX.
+It additionally provides a variety of built-in transitions for common effects, support for parallel and sequential transitions, and the ability to handle events upon animation completion.
+I will go through all types of animations, starting with `Animation` and its subclasses `Transition` and `Timeline`, before representing a lower level animation with `AnimationTimer`.
+While `Transition` provides a simpler and more user-friendly way to create animations, `Timeline` offers greater flexibility and is suitable for more complex animations.
+In contrast, `AnimationTimer` is designed for frame-by-frame updates and does not make use of `WritableValue<T>`.
+
+## WritableValue
+`WritableValue<T>` is an interface that wraps a value that can be read and set. It is commonly used for storing properties in JavaFX UI elements, like
+`width` or `height` in the [Rectangle](javafxdoc:Rectangle) shape.
+
+## Animation
+
+The [Animation](javafxdoc:Animation) abstract class provides the core functionality for `Transition` and `Timeline` animations.
```

Comment:
Every time a class name is used, it should be in code font (same for a few other occurrences):

```suggestion
The abstract class [`Animation`](javafxdoc:Animation) provides the core functionality for `Transition` and `Timeline` animations.
```

Reply:
Can you add a (half) sentence explaining that `Animation` can only be used through these subclasses, so developers don't get sidetracked into thinking about or even trying to create their own subclasses?

Reply:
> Every time a class name is used, it should be in code font (same for a few other occurrences):

What about fields and method references? 
For example, `Interpolator.DISCRETE` in the `Interpolator` section.

Reply:
Same. But there is some gray area when talking about a concept with the same name as a class, e.g.:

> The animation takes 2s.

versus

> The `Animation` has a method that...